workers/pipeline/git.go

```diff
@@ -269,6 +275,9 @@ func getAuthInfo(repo *gaia.GitRepo) (transport.AuthMethod, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// Ignore host key check for the user experience
+		auth.(*ssh.PublicKeys).HostKeyCallback = ssh2.InsecureIgnoreHostKey()
```

Comment:
@michelvocks  I would advice against doing this. If it's a public repo, you should use https... other wise, you should add the host to hostkeys. I didn't want to see something like this in production, don't you think? :)

Reply:
We should validate known_hosts file or use FixedHostKey

Reply:
@Skarlso This relates to SSH git clone. SSH doesn't use HTTPS since it has its own protocol. Usually, the SSH agent checks if this host is known to prevent man-in-the-middle attacks.

I personally think we can go without this because it makes the setup inside a docker container way more complicated. Additionally, this also the default behavior of Jenkins: https://support.cloudbees.com/hc/en-us/articles/115000073552-Host-Key-Verification-for-SSH-Agents

Reply:
What I'm saying is if the host is unknown or insecure then rather then using git@ the user should use https or add the key to known_hosts. Also jenkins is a festering pile of security issues. Every week there is a new issue. 😂 I wouldn't use them as an example of good practice regarding security, don't you think?

Reply:
Ahoi! What would be a clean solution then? I guess we would require user interaction at this point (so more or less "copy the console behavior"), asking for permission, add the host. Maybe it would be a good idea to configure the container to use the known hosts not under `~/.ssh/known_hosts` but on a path of the application, which the user might want to mount anyways (e.g. the user might want to mount a volume on the pipeline-config directory to keep the state).

Reply:
Hm, plus would need error handling in case the host ip changes.. meh.

Reply:
I would suggest looking into hostkey verification. The ssh package has this: [ssh.ParseKnownHosts](https://godoc.org/golang.org/x/crypto/ssh#ParseKnownHosts) which might provide some insight. You could also parse out host keys and then use FixedHostKey like this:

```
hostKey := getHostKey(url)
...
HostKeyCallback: ssh.FixedHostKey(hostKey),
```

There are several posts around that talk about SSH seucirty and how checking host source is important so I won't go into that here.

Some links:

https://utcc.utoronto.ca/~cks/space/blog/programming/GoSSHHostKeyCheckingNotes
https://bridge.grumpy-troll.org/2017/04/golang-ssh-security/

Reply:
To be honest, I don't think is really a security issue:

```
I’ve yet to find a project which actually bothered implementing hostkey verification in SSH client code. None.
```

This actually confirms my acceptance. We should document that for sure.

Reply:
But the fact that they don't isn't a good thing. :D And we good lead by good example. :))

But of course it's up to you man. :) I realise that this isn't trivial.

Reply:
@michelvocks @leberknecht  Hey guys. I wrote this thing: https://gist.github.com/Skarlso/34321a230cf0245018288686c9e70b2d

If the entry is hostname based this will work even if the IP changes underneath and there is no new entry for the new IP. This will also fail if there is no entry in the known_hosts file.

This can be easily modified to automatically add one for example... But I would prefer the user managing the host file instead of gaia doing it automatically.

What say you?

Reply:
I think i would be fine with a speaking error message and a small "how to configure the known hosts for containers / k8s" hint in the documentation :)

Even though it could feel a bit "unusual" to the user (as in "its normal to configure the ssh key for cloning somewhere, as its in the nature of things, but i never needed to configure the known-hosts on any other software before"). But i totally agree that copying "bad behavior from other software" is semi cool :) 

What would be the downside of doing the automatic handling? I guess that we presumably take user agreement to modify the known hosts as given?

Reply:
You never had to because at some point in time you had a question appear when you first cloned something that asked if you would like to add git to your host file. Try removing git from your known hosts and clone something. It will ask you to add it.

You indirectly configure it. 🙂

Reply:
The problem, which I have personally experienced, is when you have your Git server behind a load balancer and when you have immutable infrastructure. 

Gitlab, for example, allows you to run it HA. Which basically means you have multiple server and any server can randomly handle the traffic. If you combine that with immutable infrastructure (e.g. when you upgrade your server or your software which is running on your server) you basically screwed because each time you replace your server it will have a different host key. 

In my opinion, SSH host key validation is important and right, but it is simply nowadays with the Cloud not applicable anymore. 

@Skarlso  @leberknecht What do you think about checking the know_hosts file and if there is something suspicious we will prompt a message in the UI. This is still quite user-friendly in my opinion.

Reply:
A fair point. :) Okay, I'll agree to that. Implementing a hook which updates the known_hosts file whenever there is a new entry for a given host might be a bit out of the scope I guess. :D 👍

Reply:
BTW in an HA environment, for a given server family you just copy the SSH certificate to the other servers.
These two:
```
/etc/ssh/ssh_host_rsa_key
/etc/ssh/ssh_host_rsa_key.pub
```

This way once a load balancer is added to known_hosts everything underneath it will forever work. The load balancer, when it's rotating in a new instance will automatically copy these keys to that instance. Once that's done, that server will also work without the need to change the known_hosts file.

Reply:
Hang on. I think I have a suitable fix for this. Gonna push up something soon.